src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoriesTreePage.tsx

```diff
@@ -46,15 +62,61 @@ const CategoriesTreePage: FC = () => {
   };
 
   const addCategory = (parentCode: string, onCreate: () => void) => {
-    setNewCategory({parentCode, onCreate});
+    setCategoryToCreate({parentCode, onCreate});
     openNewCategoryModal();
   };
 
   const handleCloseNewCategoryModal = () => {
-    setNewCategory(null);
+    setCategoryToCreate(null);
     closeNewCategoryModal();
   };
 
+  const confirmDeleteCategory = async (identifier: number, label: string, onDelete: () => void) => {
+    // @todo consider the risk of this call being too long (modal not display until the response)
+    const numberOfProducts = await useCountNumberOfProductsByCategory(identifier);
```

Comment:
By convention, we use to prefix the React hooks by use. Here we do not you it like a hook
```suggestion
    const numberOfProducts = await countNumberOfProductsByCategory(identifier);
```